test/Royalties.t.sol

```diff
@@ -0,0 +1,792 @@
+        assert(!gogeToken.isExcludedFromFees(address(this)));
+        uint256 preBal = gogeToken.balanceOf(address(joe));
+
+        uint256 sendAmt = 1_000_000;
+
+        gogeToken.transfer(address(joe), sendAmt);
+
+        // Grab post balanace and calc amount goge tokens received.
+        uint256 postBal        = gogeToken.balanceOf(address(joe));
+        uint256 amountReceived = (postBal - preBal);
+        uint256 taxedAmount    = sendAmt * gogeToken.totalFees()/100;
+
+        // Verify the quoted amount is the amount received and no royalties were generated.
+        assertEq(sendAmt - taxedAmount, amountReceived);
+        assertEq(IERC20(address(gogeToken)).balanceOf(address(gogeToken)), taxedAmount);
+
+        // Log
+        emit log_uint(sendAmt);
+        emit log_uint(amountReceived);
+        emit log_uint(IERC20(address(gogeToken)).balanceOf(address(gogeToken)));
+    }
+
+    // verify whitelisted transfer
+    function test_royaltyTesting_transfer_noTax() public {
+        // Verify address(this) is NOT excluded from fees and grab pre balance.
+        assert(gogeToken.isExcludedFromFees(address(this)));
+        uint256 preBal = gogeToken.balanceOf(address(joe));
+
+        uint256 sendAmt = 1_000_000;
+
+        gogeToken.transfer(address(joe), sendAmt);
+
+        // Grab post balanace and calc amount goge tokens received.
+        uint256 postBal        = gogeToken.balanceOf(address(joe));
+        uint256 amountReceived = (postBal - preBal);
+
+        // Verify the quoted amount is the amount received and no royalties were generated.
+        assertEq(amountReceived, sendAmt);
+        assertEq(IERC20(address(gogeToken)).balanceOf(address(gogeToken)), 0);
+
+        // Log
+        emit log_uint(sendAmt);
+        emit log_uint(amountReceived);
+        emit log_uint(IERC20(address(gogeToken)).balanceOf(address(gogeToken)));
+    }
+
+
+    // ~~ Fee Generation and Tax Distributions ~~
+
+
+    // NOTE: swapTokensAtAmount distribution threshold = 20_000_000
+    //       bnb = $290.11
+    function test_royaltyTesting_generateFees() public {
+        // Remove address(this) from whitelist so we can yield a buy tax.
+        gogeToken.excludeFromFees(address(this), false);
+
+        // Check balance of address(gogeToken) to see how many tokens have been taxed. Should be 0
+        assertEq(IERC20(address(gogeToken)).balanceOf(address(gogeToken)), 0);
+
+        // Generate a buy - log amount of tokens accrued
+        buy_generateFees(1 ether);
+        emit log_uint(IERC20(address(gogeToken)).balanceOf(address(gogeToken))); // 7_901_185
+
+        buy_generateFees(2 ether);
+        emit log_uint(IERC20(address(gogeToken)).balanceOf(address(gogeToken))); // 23_244_038
+
+        sell_generateFees(10_000 ether);
+        emit log_uint(IERC20(address(gogeToken)).balanceOf(address(gogeToken))); // 3_411
+    }
+
+    // NOTE: Error in this test.
```

Comment:
Was this a fuzzing test case previously? How did you determine these exact values?

Reply:
They WERE failing. When I ran fuzzing these cases failed. All cases are now passing so I just leave this specify test here for good measure